modules/apps/29-fee/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,160 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+)
+
+func (suite *KeeperTestSuite) TestQueryIncentivizedPacket() {
+
+	var (
+		req *types.QueryIncentivizedPacketRequest
+	)
+
+	// setup
+	validPacketId := suite.NewPacketId(uint64(1))
+	invalidPacketId := suite.NewPacketId(uint64(2))
+	coins := sdk.Coins{sdk.Coin{Denom: sdk.DefaultBondDenom, Amount: sdk.NewInt(100)}}
+	ackFee := coins
+	receiveFee := coins
+	timeoutFee := coins
+	fee := &types.Fee{
+		AckFee:     ackFee,
+		ReceiveFee: receiveFee,
+		TimeoutFee: timeoutFee,
+	}
+
+	identifiedPacketFee := suite.NewIdentifiedPacketFee(validPacketId, *fee)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {
+				req = &types.QueryIncentivizedPacketRequest{
+					PacketId:    &validPacketId,
+					QueryHeight: 0,
+				}
+			},
+			true,
+		},
+		{
+			"packetId not found",
+			func() {
+				req = &types.QueryIncentivizedPacketRequest{
+					PacketId:    &invalidPacketId,
+					QueryHeight: 0,
+				}
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+
+			refundAcc := suite.chainA.SenderAccount.GetAddress()
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.chainA.GetContext())
+			suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), refundAcc, &identifiedPacketFee)
+			res, err := suite.queryClient.IncentivizedPacket(ctx, req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+				suite.Require().Equal(&identifiedPacketFee, res.IncentivizedPacket)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestQueryIncentivizedPackets() {
+	var (
+		req        *types.QueryIncentivizedPacketsRequest
+		expPackets []*types.IdentifiedPacketFee
+	)
+
+	coins := sdk.Coins{sdk.Coin{Denom: sdk.DefaultBondDenom, Amount: sdk.NewInt(100)}}
+	ackFee := coins
+	receiveFee := coins
+	timeoutFee := coins
+	fee := &types.Fee{
+		AckFee:     ackFee,
+		ReceiveFee: receiveFee,
+		TimeoutFee: timeoutFee,
+	}
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"empty pagination",
+			func() {
+				req = &types.QueryIncentivizedPacketsRequest{}
+			},
+			true,
+		},
+		{
+			"success",
+			func() {
+				refundAcc := suite.chainA.SenderAccount.GetAddress()
+
+				id1 := suite.NewPacketId(uint64(1))
```

Comment:
You can call `NewPacketId` inside of `NewIdentifiedPacketFee` to avoid redundant variable assignments here too.
You should also be able to drop the `uint64()` conversion afaik, as the arg is defined as `uint64`

```
NewIdentifiedPacketFee(NewPacketId(1), *fee)
```